ui/pages/confirmations/hooks/alerts/signatures/useDomainMismatchAlert.ts

```diff
@@ -0,0 +1,42 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  isValidSIWEOrigin,
+  WrappedSIWERequest,
+} from '@metamask/controller-utils';
+
+import { Alert } from '../../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../../selectors';
+
+import { SignatureRequestType } from '../../../types/confirm';
+import { isSIWESignatureRequest } from '../../../utils';
+
+export default function useDomainMismatchAlerts(): Alert[] {
+  const t = useI18nContext();
+
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as SignatureRequestType;
+  const { msgParams } = currentConfirmation;
+
+  const isValidSIWE = currentConfirmation?.msgParams && isSIWESignatureRequest(currentConfirmation);
+  const isValidDomain = isValidSIWEOrigin(msgParams as WrappedSIWERequest);
+
+  return useMemo(() => {
+    if (!isValidSIWE || isValidDomain) {
```

Comment:
Apologies again, but `isValidSIWE` is not listed as a dependency, plus we shouldn't need to provide it inside the `useMemo` if we instead inverted `isValidDomain` to check it also?

`const isInvalidDomain = isValidSIWE && !isValidSIWEOrigin(msgParams as WrappedSIWERequest);`

Reply:
whoops, my apologies. how's this? 
[5ae550d](https://github.com/MetaMask/metamask-extension/pull/25281/commits/5ae550d6ce08f711b4c655c62d6d37d21df3dc8c)

I don't want to mix the context of `isValidSIWE` in `isInvalidDomain`, so I kept them separate. ended up reverting to kind of the initial useMemo solution we had except with the value calculations outside of the useMemo hook